domain/src/main/java/com/github/leuludyha/domain/useCase/SearchRemotelyUseCase.kt

```diff
@@ -1,8 +1,17 @@
 package com.github.leuludyha.domain.useCase
 
+import android.content.Context
 import com.github.leuludyha.domain.repository.LibraryRepository
+import com.github.leuludyha.domain.util.NetworkProvider
+import kotlinx.coroutines.flow.flowOf
 
-class SearchRemotelyUseCase(private val libraryRepository: LibraryRepository) {
-    operator fun invoke(query: String) =
-        libraryRepository.searchRemotely(query)
+class SearchRemotelyUseCase(
+    private val libraryRepository: LibraryRepository,
+    private val networkProvider: NetworkProvider
+) {
+    operator fun invoke(context: Context, query: String) =
+        if(networkProvider.checkNetworkAvailable(context))
+            libraryRepository.searchRemotely(query)
+        else
+            flowOf()
```

Comment:
In all other cases you return an error inside the flow in case there is no network available. I'm curious, what is the reason for not doing it here?

Reply:
My bad, it should return an error as well